modules/apps/style-book/style-book-web/src/main/resources/META-INF/resources/js/style-book-editor/LayoutSelector.js

```diff
@@ -35,6 +35,9 @@ export default function LayoutSelector() {
 			alignmentPosition={Align.BottomRight}
 			menuElementAttrs={{
 				className: 'style-book-editor__page-selector',
+				containerProps: {
+					className: 'cadmin',
+				},
```

Comment:
Hey @pat270 I've been thinking about trying to improve this implementation to avoid having to add `cadmin` to all Clay components in all products, I'm thinking about adding a `Theme` that is defined globally and adding the configured selector to all Clay components per default, so we can keep track of that within Clay and we would just need to add to the `render` that is used in most applications.

We had been thinking about creating a `ClayProvider` or just `Provider` to handle things like this https://github.com/liferay/clay/issues/2353.

Reply:
The only caveat for using a Provider or react context is that we have to account for non-react modules and server rendered tags

Reply:
Yeah, those cases would still have to add wrappers unfortunately but it would save extra work because people can easily use Clay components and forget to add scope.

Reply:
If we can make the syntax consistent between react and non-react components, it would be a major improvement. I personally like it when the syntax is consistent even if it's more verbose. If you think it will increase productivity and reduce bugs, I guess it would be good? I'm not a big fan of tags and wrappers myself, I feel like it slows me down (e.g., container-fluid tag).

Reply:
The idea is that we only configure the `theme` in one place instead of defining the scope for each component.

Reply:
My understanding of the subject is none. There would be a `theme` setting and it would wrap `style-book-editor` in `cadmin` and dropdown-menu's in `cadmin`?

Reply:
No problem, I can describe it better, for example, we will create a `ClayProvider` in React that has the `theme` property which in this case would be the typed selector, under the hood the Clay components will consume this information from the `ClayProvider` and add the selector on the components for the `DropDown`... cases that use the ReactPortal as you are doing manually here.

So any Clay component that is below Provider will have this class, we can also add this to the `render` that the `react:component` taglib uses to render most applications in React to have this default effect.

```jsx
// With Scope
<ClayProvider theme="cadmin">
  <App>
     <ClayDropDownWithItems items={...} />
  </App>
<ClayProvider/>

// No scope
<App>
  <ClayDropDownWithItems items={...} />
</App>
```

Does it make sense?

Reply:
I see makes sense.

Reply:
So I'm going to start working on that next week or this week if I have time and then we can replace these cases with this new implementation, what do you think? can this help you?

Reply:
This wouldn't change much for me. It would be more for the developer experience. I think it would be better to get feedback from the component devs since they will be interacting with it more than I will. I also think there will be cases where this pattern wouldn't work and need to fallback to adding the `cadmin` class.

Reply:
Hmm, do you have a vision of which cases this could be bad? I imagine this would avoid the problems of forgetting to add the `cadmin` selector in cases like this, because it looks a lot like the `Theme` pattern in components, usually, you can define it in the component or use it in the `Provider` of the application.

This is also similar in the same case we do with `spritemap` the developers don't have to worry about adding for every component because we add in Provider.

I probably see that this doesn't really cover all the use cases, because we still have taglibs and JSP components but it solves a lot of the modern part we have.

Reply:
It's the same thing Bryce mentioned (non-react modules and server side code). Here are some examples I could find:

https://github.com/liferay-frontend/liferay-portal/blob/821d682bb25be3232c78740d7f39240795ee941a/modules/apps/product-navigation/product-navigation-product-menu-web/src/main/java/com/liferay/product/navigation/product/menu/web/internal/servlet/taglib/ProductMenuBodyTopDynamicInclude.java#L126

https://github.com/liferay-frontend/liferay-portal/blob/821d682bb25be3232c78740d7f39240795ee941a/modules/apps/frontend-theme/frontend-theme-classic/src/templates/portlet.ftl#L23

Need to pass in cadmin through direction attribute:
https://github.com/liferay-frontend/liferay-portal/blob/821d682bb25be3232c78740d7f39240795ee941a/modules/apps/frontend-theme/frontend-theme-classic/src/templates/portlet.ftl#L30

Mixed use, where we need two themes. We can't wrap page editor drag and drop controls in `cadmin`. We rely on `html:not(__):not()___` to do the scoping:
https://github.com/liferay-frontend/liferay-portal/blob/821d682bb25be3232c78740d7f39240795ee941a/modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/Topper.js#L355

Reply:
Yeah, of course, these cases unfortunately we can't handle but we can smooth over from the side of the React components.

Reply:
Ok sounds good.